tests/e2e/login_test.go

```diff
@@ -8,91 +8,96 @@ import (
-				Expect(session).To(ContainSubstring(loginTestUser))
-				token := runCmd(ocTokenCommand)
+				session = runCmd("oc whoami")
+				Expect(session).To(ContainSubstring(loginTestUserForNoProject))
 				// One initialization needs one login, hence it happens here
-				testUserLoginCommandWithToken = fmt.Sprintf("%s -t %s", baseOdoLoginCommand, token)
+				testUserToken = runCmd("oc whoami -t")
 			})
 
 			It("Should login successfully with token without any projects with appropriate message", func() {
-				session = runCmd(testUserLoginCommandWithToken)
+				session = runCmd(fmt.Sprintf("odo login -t %s", testUserToken))
 				Expect(session).To(ContainSubstring("Logged into"))
 				Expect(session).To(ContainSubstring("You don't have any projects. You can try to create a new project, by running"))
 				Expect(session).To(ContainSubstring("odo project create <project-name>"))
-				session = runCmd(ocWhoamiCommand)
-				Expect(session).To(ContainSubstring(loginTestUser))
-			})
-
-			It("Should login successfully with username and password single project with appropriate message", func() {
-				runCmd(testUserLoginCommand)
-				runCmd(testUserCreateProject1Command)
-				runCmd(backToCurrentUserCommand)
-				session = runCmd(testUserLoginCommand)
-				Expect(session).To(ContainSubstring("Login successful"))
-				Expect(session).To(ContainSubstring(odoTestProject1))
-				session = runCmd(ocWhoamiCommand)
-				Expect(session).To(ContainSubstring(loginTestUser))
+				session = runCmd("oc whoami")
+				Expect(session).To(ContainSubstring(loginTestUserForNoProject))
 			})
 
 			It("Should fail login on invalid token with appropriate message", func() {
-				session = runFailCmd(testUserLoginFailCommandWithToken, 1)
+				session = runFailCmd("odo login -t verybadtoken", 1)
 				Expect(session).To(ContainSubstring("The token provided is invalid or expired"))
-				runCmd(testUserLoginCommand)
+				runCmd(fmt.Sprintf("oc login --token %s", testUserToken))
+			})
+		})
+
+		Context("Run login tests with single active project with username and password", func() {
+			AfterEach(func() {
+				cleanUpAfterProjects([]string{odoTestProjectForSingleProject1})
+			})
+
+			It("Should login successfully with username and password single project with appropriate message", func() {
+				// Initialise for test
+				runCmd(fmt.Sprintf("oc login -u %s -p %s", loginTestUserForSingleProject1, loginTestUserPassword))
+				runCmd(fmt.Sprintf("odo project create %s", odoTestProjectForSingleProject1))
+				runCmd("oc logout")
+				session = runCmd(fmt.Sprintf("odo login -u %s -p %s", loginTestUserForSingleProject1, loginTestUserPassword))
+				Expect(session).To(ContainSubstring("Login successful"))
+				Expect(session).To(ContainSubstring(odoTestProjectForSingleProject1))
+				session = runCmd("oc whoami")
+				Expect(session).To(ContainSubstring(loginTestUserForSingleProject1))
 			})
 		})
 	})
 })
+
+func cleanUpAfterProjects(projects []string) {
```

Comment:
Can you keep function ```cleanUpAfterProjects``` in e2e.go file just to make sure the consistency of the test structure.

Reply:
Done